src/applications/static-pages/i18Select/hooks.js

```diff
@@ -26,22 +26,30 @@ export const setLangAttribute = lang => {
 // without overwriting existing onclick events
 // reference: https://stackoverflow.com/questions/891989/javascript-adding-an-onclick-handler-without-overwriting-the-existing-one
 
-export const adaptLinksWithLangCode = setLangAttributeInReduxStore => {
+export const adaptLinksWithLangCode = (
+  setLangAttributeInReduxStore,
+  languageCodeInReduxStore,
+) => {
   const links = document.links;
   for (const link of links) {
-    link.addEventListener('click', () => {
-      setLangAttributeInReduxStore('en');
-      const langAttribute = link.hreflang || link.lang;
-      if (langAttribute) {
-        setLangAttributeInReduxStore(langAttribute);
-      }
-      // respect the temp IA i18 structure
-      if (link.href.endsWith('-esp/')) {
-        setLangAttributeInReduxStore('es');
-      }
-      if (link.href.endsWith('-tag/')) {
-        setLangAttributeInReduxStore('tl');
-      }
-    });
+    if (link) {
+      link.setAttribute('lang', languageCodeInReduxStore);
+      link.setAttribute('hreflang', languageCodeInReduxStore);
+
+      link.addEventListener('click', () => {
+        setLangAttributeInReduxStore('en');
+        const langAttribute = link.lang || link.hreflang;
+        if (langAttribute) {
+          setLangAttributeInReduxStore(langAttribute);
```

Comment:
Do you ever a race conditions here where `setLangAttributeInReduxStore('en');` is called and set after a `setLangAttributeInReduxStore(langAttribute);`?

Reply:
good point! I think redux has its own method of internally handling callbacks, but i haven't run into it just yet and will did further. the default setting of `en` lang code _**may**_ not be necessary